libs/cimit-service/src/main/java/uk/gov/di/ipv/core/library/service/CiMitService.java

```diff
@@ -157,57 +255,79 @@ public ContraIndicators getContraIndicators(VerifiableCredential vc)
                 .build();
     }
 
+    @Tracing
     public VerifiableCredential getContraIndicatorsVc(
             String userId, String govukSigninJourneyId, String ipAddress)
             throws CiRetrievalException {
         var response = invokeClientToGetCIResult(govukSigninJourneyId, ipAddress, userId);
 
         try {
             ContraIndicatorCredentialDto contraIndicatorCredential =
-                    OBJECT_MAPPER.readValue(
-                            response.payload().asUtf8String(), ContraIndicatorCredentialDto.class);
+                    OBJECT_MAPPER.readValue(response, ContraIndicatorCredentialDto.class);
             return extractAndValidateContraIndicatorsJwt(contraIndicatorCredential.getVc(), userId);
         } catch (JsonProcessingException e) {
             throw new CiRetrievalException("Failed to deserialize ContraIndicatorCredentialDto");
         }
     }
 
-    private InvokeResponse invokeClientToGetCIResult(
+    private String invokeClientToGetCIResult(
             String govukSigninJourneyId, String ipAddress, String userId)
             throws CiRetrievalException {
         LOGGER.info(LogHelper.buildLogMessage("Retrieving CIs from CIMIT system"));
-
-        String payload;
-        try {
-            payload =
-                    OBJECT_MAPPER.writeValueAsString(
-                            new GetCiRequest(govukSigninJourneyId, ipAddress, userId));
-        } catch (JsonProcessingException e) {
-            throw new CiRetrievalException("Failed to serialize GetCiRequest");
-        }
-
-        var invokeRequest =
-                InvokeRequest.builder()
-                        .functionName(
-                                configService.getEnvironmentVariable(
-                                        CIMIT_GET_CONTRAINDICATORS_LAMBDA_ARN))
-                        .payload(SdkBytes.fromUtf8String(payload))
-                        .qualifier(LIVE_ALIAS)
-                        .build();
-
-        InvokeResponse response;
         try {
-            response = lambdaClient.invoke(invokeRequest);
+            if (configService.enabled(CIMIT_API_GATEWAY_ENABLED)) {
+                var uriBuilder = getUriBuilderWithBaseApiUrl(GET_VCS_ENDPOINT);
+
+                var uri = uriBuilder.addParameter(USER_ID_PARAMETER, userId).build();
+
+                var response = sendGetHttpRequest(uri, govukSigninJourneyId, ipAddress);
```

Comment:
(similarly - maybe we should just add `userId` as a parameter to this method and pass along with `GET_VCS_ENDPOINT`, so the URL is built in one place)